cmd/soroban-rpc/internal/test/metrics_test.go

```diff
@@ -25,6 +28,23 @@ func TestMetrics(t *testing.T) {
 		config.Version,
 	)
 	require.Contains(t, metrics, buildMetric)
+
+	logger := test.daemon.Logger()
+	err := errors.Errorf("test-error")
+	logger.WithError(err).Error("test error 1")
+	logger.WithError(err).Error("test error 2")
+
+	metricFamilies, err := test.daemon.MetricsRegistry().Gather()
+	assert.NoError(t, err)
+	var metric *io_prometheus_client.MetricFamily
+	for _, mf := range metricFamilies {
+		if *mf.Name == "soroban_rpc_log_error_total" {
+			metric = mf
+			break
+		}
+	}
+	val := metric.Metric[0].Counter.GetValue()
```

Comment:
may want to assert that `metric` is non-nil first to emit a valid assertion error instead of potentially just a nil pointer panic.